pkg/executor/aggregate/agg_util.go

```diff
@@ -40,6 +41,43 @@ import (
 	"go.uber.org/zap"
 )
 
+const defaultPartialResultsBufferCap = 2048
+const defaultGroupKeyCap = 8
+
+var partialResultsBufferPool = sync.Pool{
+	New: func() any {
+		s := make([][]aggfuncs.PartialResult, 0, defaultPartialResultsBufferCap)
+		return &s
+	},
+}
+
+var groupKeyPool = sync.Pool{
+	New: func() any {
+		s := make([][]byte, 0, defaultGroupKeyCap)
+		return &s
+	},
+}
+
+func getBuffer() (*[][]aggfuncs.PartialResult, *[][]byte) {
```

Comment:
Why use `*[][]xx` instead of `[][]xx`

Reply:
> Why use `*[][]xx` instead of `[][]xx`

It's better to return pointer type in sync.Pool, [refer](https://stackoverflow.com/questions/54179128/is-it-necessary-to-return-pointer-type-in-sync-pool-new-function)